File: Ex9.java
package Ex9;

interface Imprimivel {
    void imprimir();
}

class Relatorio implements Imprimivel {
    private String conteudo;

    public Relatorio(String conteudo) {
        this.conteudo = conteudo;
    }

    @Override
    public void imprimir() {
        System.out.println("Imprimindo Relatório:");
        System.out.println(conteudo);
    }
}

class Contrato implements Imprimivel {
    private String termos;

    public Contrato(String termos) {
        this.termos = termos;
    }

    @Override
    public void imprimir() {
        System.out.println("Imprimindo Contrato:");
        System.out.println(termos);
    }
}

public class Ex9 {
    public static void main(String[] args) {
        Relatorio relatorio = new Relatorio("Relatório de Vendas do Mês.");
        Contrato contrato = new Contrato("Contrato de Prestação de Serviços.");

        relatorio.imprimir();
        contrato.imprimir();
    }
}

File: ex9.go
package main

import (
	"fmt"
)

type Imprimivel interface {
	Imprimir()
}

type Relatorio struct {
	Titulo   string
	Conteudo string
}

func (r Relatorio) Imprimir() {
	fmt.Printf("Relatório: %s\nConteúdo: %s\n", r.Titulo, r.Conteudo)
}

type Contrato struct {
	Titulo string
	Valor  float64
}

func (c Contrato) Imprimir() {
	fmt.Printf("Contrato: %s\nValor: R$ %.2f\n", c.Titulo, c.Valor)
}

func ExibirImprimivel(i Imprimivel) {
	i.Imprimir()
}

func main() {
	relatorio := Relatorio{Titulo: "Relatório Anual", Conteudo: "Este é o conteúdo do relatório."}
	contrato := Contrato{Titulo: "Contrato de Prestação de Serviços", Valor: 5000.00}

	ExibirImprimivel(relatorio)
	ExibirImprimivel(contrato)
}
